test_critical_fixes.py

```
@@ -0,0 +1,373 @@
+        async with get_session() as session:
+            # Simple query
+            result = await session.execute(text("SELECT 1"))
+            assert result.scalar() == 1
+            print("✓ Database session created successfully")
+
+        # Test error handling (should use SQLAlchemyError, not Exception)
+        try:
+            async with get_session() as session:
+                # This will trigger rollback
+                await session.execute(text("INVALID SQL"))
+        except Exception as e:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)